src/main/scala/co/teapot/tempest/server/TempestDatabaseClient.scala

```diff
@@ -220,6 +225,49 @@ class TempestSQLDatabaseClient(config: DatabaseConfig) extends TempestDatabaseCl
     withConnection { implicit connection =>
       SQL(s"INSERT INTO ${nodesTable(node.`type`)} (id) VALUES ({id})")
         .on("id" -> node.id).execute()
+    }
+
+  def addNodes(nodes: Seq[ThriftNode]): Unit = {
+    if (nodes.isEmpty)
+      return
+
+    try {
+      // Currently, only support adding nodes of 1 type in a batch
+      val nodeTypes = nodes.map(_.`type`).toSet
+
+      if (nodeTypes.size > 1)
+        throw new InvalidArgumentException(s"All nodes need to be of the same type. Encountered node types: $nodeTypes")
+
+      val nodeIds = nodes.map(_.id).distinct
+      val tableName = nodesTable(nodes.head.`type`)
+
+      withConnection { implicit connection =>
+        val parameters = nodeIds.map { id => Seq[NamedParameter]("id" -> id) }
+
+        // Separating into head and tail is weird, but for some reason the anorm library wants it
+        // this way
+        val query = BatchSql(s"INSERT INTO $tableName (id) VALUES({id})",
+          parameters.head, parameters.tail:_*
+        )
+
+        query.execute()
+      }
+    } catch {
+      case e: BatchUpdateException => throw new SQLException(e.getMessage)
+    }
+  }
+
+  def addNewNodes(nodes: Seq[ThriftNode]): Unit = {
+    if (nodes.isEmpty)
+      return
+
+    val nodeIds = nodes.map(_.id)
+    val nodeType = nodes.head.`type`
+
+    val existingNodeIds = filterNodeIds(nodeType, nodeIds, "1 = 1").toSet
```

Comment:
"1 = 1" here feels weird.  If you have time, would replacing it with "TRUE" work?